src/main/java/domain/AttendanceStatistics.java

```diff
@@ -0,0 +1,57 @@
+package domain;
+
+import java.time.LocalDate;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class AttendanceStatistics {
+
+    private static final int COUNT_START_YEAR = 2024;
+    private static final int COUNT_START_MONTH = 12;
+    private static final int COUNT_START_DAY = 2; // TODO:주말 처리하고 1로 바꾸기
+
+    public static StatisticsResult countStatus(LocalDate nowDate, Records records) {
+        int attendanceCount = 0;
+        int latenessCount = 0;
+        int absenceCount = 0;
+
+        LocalDate startDate = LocalDate.of(COUNT_START_YEAR, COUNT_START_MONTH, COUNT_START_DAY);
+        while (startDate.isBefore(nowDate)) {
+            TimeAndStatus status = records.findByDate(startDate);
+            startDate = startDate.plusDays(1);
+
+            if (!Holiday.isHoliday(startDate.minusDays(1))) {
+                if (status == null || status.getStatus() == null) {
+                    absenceCount++;
+                    continue;
+                }
+
+                if (status.getStatus().equals("출석")) {
+                    attendanceCount++;
+                }
+                if (status.getStatus().equals("지각")) {
+                    latenessCount++;
+                }
+                if (status.getStatus().equals("결석")) {
+                    absenceCount++;
+                }
+            }
+        }
+        return new StatisticsResult(attendanceCount, latenessCount, absenceCount);
+    }
+
+    public static Map<String, StatisticsResult> calculateExpelledWarning(LocalDate nowDate,
+        Map<String, Records> crews) {
+        Map<String, StatisticsResult> result = new LinkedHashMap<>();
+        for (String crewName : crews.keySet()) {
+            StatisticsResult statisticsResult = AttendanceStatistics.countStatus(nowDate,
+                crews.get(crewName));
+
+            Penalty penalty = statisticsResult.getPenalty();
+            if (penalty != Penalty.NONE) {
+                result.put(crewName, statisticsResult);
+            }
```

Comment:
```suggestion
            if (statisticsResult.hasPenalty()) {
                result.put(crewName, statisticsResult);
            }
```

- 값을 가져와서 검증
- 객체에게 검증로직 호출

어떤 차이가 있을까요?

Reply:
시간이 부족해서 getter를 사용하게 됐는데, 사용하면서도 개선 방법이 존재할 것이라는 생각이 들었습니다.

객체에게 검증로직을 호출하도록 하면 객체가 스스로 검증해서 결과만 알려주게 되어 getter를 사용하지 않도록 개선 가능해보입니다! 리팩터링에 반영했습니다 😊

Reply:
참고하면 좋을 자료 공유드립니다~

- https://tecoble.techcourse.co.kr/post/2020-04-28-ask-instead-of-getter/